### PR TITLE
Feat/dx 544 add boilerplates

### PR DESCRIPTION
**Pull request by harshithad0703**
- [x] add more boilerplate template apps
- [x] version bump 1.2.1 -> 1.2.2

### CODE REVIEW

File: src/commands/app/create.ts [hunk 1]
```diff
@@ -96,7 +97,15 @@ export default class Create extends BaseCommand<typeof Create> {
           message: this.messages.CONFIRM_CLONE_BOILERPLATE,
         }))
       ) {
-        await this.boilerplateFlow();
+        const boilerplate = await selectedBoilerplate();
+        if (boilerplate) {
+          this.sharedConfig.boilerplateName = boilerplate.name
+            .toLowerCase()
+            .replace(/ /g, "-");
```
Comment (aman19K) [inline src/commands/app/create.ts:104]: @harshithad0703 , Please review the previous flow. We were asking user to input app name using the prompt; you don't have to create it. Here you can check the condition **contentstack-apps-cli/src/util/inquirer.ts**

File: src/commands/app/create.ts [hunk 2]
```diff
@@ -96,7 +97,15 @@ export default class Create extends BaseCommand<typeof Create> {
           message: this.messages.CONFIRM_CLONE_BOILERPLATE,
         }))
       ) {
-        await this.boilerplateFlow();
+        const boilerplate = await selectedBoilerplate();
```
Comment (aman19K) [inline src/commands/app/create.ts:100]: @harshithad0703 , You are attempting to use boilerplate directly below, and the boilerplate return type is any.The entire process will break if the boilerplate is an empty object.

File: src/commands/app/create.ts [hunk 3]
```diff
@@ -236,8 +259,8 @@ export default class Create extends BaseCommand<typeof Create> {
   manageManifestToggeling() {
     // NOTE Use boilerplate manifest if exist
     const manifestPath = resolve(
-      this.sharedConfig.folderPath || "",
-      "manifest.json"
+      this.sharedConfig.folderPath,
+      `${this.sharedConfig.folderPath}/manifest.json`
     );
```
Comment (aman19K) [inline src/commands/app/create.ts:264]: @harshithad0703 , add check if manifest file not present in repo then use static one.

**Review by aman19K — CHANGES_REQUESTED** [review]

File: src/config/manifest.json
[the remark this answers is not in the document]
Comment (harshithad0703) [reply src/config/manifest.json:1]: added the manifest file

File: src/util/common-utils.ts [hunk 4]
```diff
@@ -396,6 +398,15 @@ const handleProjectNameConflict = async (
   }
   return projectName;
 };
+async function fetchBoilerplateDetails(): Promise<Record<string, any>[]> {
+  try {
+    const url = config.boilerplatesUrl;
+    const content = await axios.get(url);
```
[the remark this answers is not in the document]
Comment (harshithad0703) [reply src/util/common-utils.ts]: used httpClient call

File: src/util/common-utils.ts [hunk 5]
```diff
@@ -396,6 +398,15 @@ const handleProjectNameConflict = async (
...
+    return content.data.templates
```
[the remark this answers is not in the document]
Comment (harshithad0703) [reply src/util/common-utils.ts]: used optional chaining

File: src/config/index.ts [hunk 6]
```diff
@@ -2,12 +2,13 @@ import { resolve } from "path";
 
 const config = {
   defaultAppName: "app-boilerplate",
-  manifestPath: resolve(__dirname, "manifest.json"),
+  manifestPath: resolve(__dirname, ""),
```
[the remark this answers is not in the document]
Comment (harshithad0703) [reply src/config/index.ts]: if the manifest.json is not present in unzipped folder then it uses the static one

File: src/util/inquirer.ts [hunk 7]
```diff
@@ -385,6 +386,22 @@ function inquireRequireValidation(input: any): string | boolean {
   return true;
 }
 
+const selectedBoilerplate = async (): Promise<any> => {
+  const boilerplates = await fetchBoilerplateDetails();
+
+  const response = await cliux
```
[the remark this answers is not in the document]
Comment (harshithad0703) [reply src/util/inquirer.ts]: returned direct call

File: src/commands/app/create.ts (hunk 2 above)
Comment (harshithad0703) [reply src/commands/app/create.ts:100]: added conditions on if

File: src/commands/app/create.ts (hunk 3 above)
Comment (harshithad0703) [reply src/commands/app/create.ts]: the existSync will check and add the the static manifest.json

File: src/commands/app/create.ts [hunk 8]
```diff
@@ -120,6 +132,12 @@ export default class Create extends BaseCommand<typeof Create> {
     await this.unZipBoilerplate(await this.cloneBoilerplate());
     tmp.setGracefulCleanup(); // NOTE If graceful cleanup is set, tmp will remove all controlled temporary objects on process exit
 
+    // To remove the default app name from flag and replace it with the actual folder name
+    this.sharedConfig.appName =
+      this.sharedConfig.folderPath.split("/").pop() ||
+      this.sharedConfig.appName;
+    this.tempAppData.name = this.sharedConfig.appName;
```
Comment (aman19K) [inline src/commands/app/create.ts:139]: @harshithad0703, pls verify and remove this code if not required.

**Review by aman19K — APPROVED** [review]

File: package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@contentstack/apps-cli",
-  "version": "1.2.1",
+  "version": "1.2.2",
```
Comment (aman19K) [inline package.json:3]: @harshithad0703 , minor version bump as it's enhancement

File: src/commands/app/create.ts (hunk 8 above)
Comment (harshithad0703) [reply src/commands/app/create.ts]: removed these lines

File: package.json (hunk 9 above)
Comment (harshithad0703) [reply package.json]: version bump to 1.3.0

**Review by aman19K — COMMENTED** [review]
approved

**Pull request merged**